src/MagnumPlugins/WebPImageConverter/Test/WebPImageConverterTest.cpp

```diff
@@ -0,0 +1,394 @@
+    CORRADE_VERIFY(converted);
+
+    CORRADE_COMPARE(converted->size(), Vector2i(2, 3));
+    CORRADE_COMPARE(converted->format(), PixelFormat::RGB8Unorm);
+
+    /* The image has four-byte aligned rows, clear the padding to deterministic
+       values */
+    CORRADE_COMPARE(converted->mutableData().size(), 24);
+    converted->mutableData()[6] = converted->mutableData()[7] =
+        converted->mutableData()[14] = converted->mutableData()[15] =
+            converted->mutableData()[22] = converted->mutableData()[23] = 0;
+
+    CORRADE_COMPARE_AS(converted->data(), Containers::arrayView(ConvertedRgb8Data_1),
+        TestSuite::Compare::Container);
+}
+
+constexpr const char OriginalRgbData16[80] = {
+    /* Skip */
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+
+    10, 20, 30, 10, 20, 30, 0, 0, 0, 0,
+    40, 50, 60, 40, 50, 60, 0, 0, 0, 0,
+    70, 80, 90, 70, 80, 90, 0, 0, 0, 0
+};
+
+const ImageView2D OriginalRgb16{PixelStorage{}.setSkip({0, 1, 0}),
+    PixelFormat::RGB16Unorm, {2, 3}, OriginalRgbData16};
+
+constexpr const char ConvertedRgb8Data_2[] = {
+    52, 52, 36, 2, 2, 0, 0, 0,
+    9, 3, 3, 60, 53, 53, 0, 0,
+    24, 13, 22, 16, 5, 13, 0, 0
+};
+
+void WebPImageConverterTest::rgb16() {
+    Containers::Optional<Containers::Array<char>> data = _converterManager.instantiate("WebPImageConverter")->convertToData(OriginalRgb16);
+    CORRADE_VERIFY(data);
+
+    if(_importerManager.loadState("WebPImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("WebPImporter plugin not found, cannot test");
+
+    Containers::Pointer<AbstractImporter> importer = _importerManager.instantiate("WebPImporter");
+    CORRADE_VERIFY(importer->openData(*data));
+    Containers::Optional<Trade::ImageData2D> converted = importer->image2D(0);
+    CORRADE_VERIFY(converted);
+
+    CORRADE_COMPARE(converted->mutableData().size(), 24);
+    converted->mutableData()[6] = converted->mutableData()[7] =
+        converted->mutableData()[14] = converted->mutableData()[15] =
+            converted->mutableData()[22] = converted->mutableData()[23] = 0;
+    CORRADE_COMPARE(converted->size(), Vector2i(2, 3));
+    CORRADE_COMPARE(converted->format(), PixelFormat::RGB8Unorm);
+    CORRADE_COMPARE_AS(converted->data(), Containers::arrayView(ConvertedRgb8Data_2),
+        TestSuite::Compare::Container);
+}
+
+constexpr const char OriginalRgbaData16[96] = {
+        /* Skip */
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0,0,
+        0,0, 0,0, 0, 0, 0, 0,
+
+        10, 20, 30, 80, 10, 20, 30, 80,
+        40, 50, 60, 90, 40, 50, 60, 90,
+        70, 80, 90, 100,70, 80, 90, 100,
+};
+
+const ImageView2D OriginalRgba16{PixelStorage{}.setSkip({0, 1, 0}).setRowLength(3),
+                                PixelFormat::RGBA16Unorm, {2, 3}, OriginalRgbaData16};
+
+constexpr const char ConvertedRgba8Data[] = {
+    70, 80, 90, 100, 70, 80, 90, 100,
+    40, 50, 60, 90, 40, 50, 60, 90,
+    10, 20, 30, 80, 10, 20, 30, 80
```

Comment:
Scan line order from the [webp sepcification](https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification):

> A processing order of pixels (left to right and top to bottom), starting from the left-hand-top pixel. Once a row is completed, continue from the left-hand column of the next row.

If that does indeed need to be Y-flipped, flipping the image like `PngImageConverter` does before encoding would work, right?

Reply:
That comment may have been confusing, sorry, updated it. WebP stores images Y down, while the in-memory data in Magnum is Y up (because they're all imported that way, currently). So the exporter has to flip back, yes.

In case of libpng it's easier because the encoder supports supplying data row-by-row. WebP doesn't, and doesn't have any explicit option for this, so it has to be either

> [...] maybe passing a pointer to the last row instead of the first and a negative stride could work 

as I said above, or making a copy of the input data and running [Utility::flipInPlace()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility.html#ab19ea7faa970d7f31bdab70097fdc291) on it before passing it to libwebp.